int16 x1
int16 x2
int16 x3
int16 x4
